src/mds/mdstypes.h

```diff
@@ -415,41 +415,37 @@ inline bool operator==(const client_writeable_range_t& l,
 
 struct inline_data_t {
 private:
-  bufferlist *blp;
+  std::unique_ptr<bufferlist> blp;
 public:
   version_t version;
 
   void free_data() {
-    delete blp;
-    blp = NULL;
+    blp.reset();
   }
   bufferlist& get_data() {
     if (!blp)
-      blp = new bufferlist;
+      blp.reset(new bufferlist);
     return *blp;
   }
   size_t length() const { return blp ? blp->length() : 0; }
 
-  inline_data_t() : blp(0), version(1) {}
-  inline_data_t(const inline_data_t& o) : blp(0), version(o.version) {
+  inline_data_t() : version(1) {}
+  inline_data_t(const inline_data_t& o) : version(o.version) {
     if (o.blp)
-      get_data() = *o.blp;
-  }
-  ~inline_data_t() {
-    free_data();
+      *blp.get() = *o.blp.get();
```

Comment:
Oops, didn't spot this until I ran the tests again, there's another crasher here, blp may be null at this point (shoudl use get_data() like before).  Same  in the copy constructor.